src/AppBundle/Service/ReportService.php

```diff
@@ -104,18 +104,67 @@ private function createNextYearReport(Report $oldReport)
             false
         );
 
+        $this->clonePersistentResources($newReport, $oldReport);
+
+        $newReport->updateSectionsStatusCache($newReport->getAvailableSections());
+        $this->_em->persist($newReport);
+
+        return $newReport;
+    }
+
+    /**
+     * Clone resources which cross report periods from one account to another
+     *
+     * @param Ndr|Report $toReport
+     * @param Ndr|Report $fromReport
+     */
+    public function clonePersistentResources($toReport, $fromReport)
+    {
         // copy assets
-        $newReport->setNoAssetToAdd($oldReport->getNoAssetToAdd());
-        foreach ($oldReport->getAssets() as $asset) {
-            $newAsset = clone $asset;
-            $newAsset->setReport($newReport);
-            $this->_em->detach($newAsset);
-            $this->_em->persist($newAsset);
+        $toReport->setNoAssetToAdd($fromReport->getNoAssetToAdd());
+        foreach ($fromReport->getAssets() as $asset) {
+            // Check that the target report doesn't already have a matching asset
+            $assetExists = false;
+            foreach ($toReport->getAssets() as $toAsset) {
```

Comment:
Out of curiosity - are these existence checks for the scenario of a resubmitted report? I'm assuming in the case of a first submit, the new report will have no assets and never meet this condition?

Reply:
Yep. The genesis of this work is that we need to copy the assets/accounts across when resubmitting reports, but we don't want to duplicate things that were created on first submit or mess with data that's been changed since.

So it will only clone the records across if they're definitely new, and won't edit or delete anything.

Reply:
LGTM